### PR TITLE
AT-16 Frontend Student Page

### PR DESCRIPTION
**Pull request by themegatree**
Page showing list of students that can be sorted alphabetically (ascending and descending).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,6 @@ yarn-debug.log*
 yarn-error.log*
 
 cypress/report
+
+#extra
+todo.md
```
Comment (dearshrewdwit) [inline .gitignore:28]: Is this relevant to the project? If it's your own personal notes, simply don't track the file.

File: cypress/integration/student.spec.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+describe("Students - display: ", function() {
+    beforeEach(function() {
+      cy.intercept("http://localhost:5000/api/students", { fixture: 'students.json' })
+    })
+
+    it("Headers correctly displayed", async function() {
+      await cy.visit('/students')
```
Comment (dearshrewdwit) [inline cypress/integration/student.spec.js:7]: what's the purpose of async await here? Cypress will usually wait long enough for any async behaviour to resolve.

File: src/components/Students/Students.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        <thead><tr>
+          <th id="lastNameHeader" onClick={(e) => {this.sortTable("student-list", 0)}} value="lastName">Last Name</th>
+          <th id="firstNameHeader"  onClick={(e) => {this.sortTable("student-list", 1)}} value="firstName">First Name</th>
+          <th id="githubUsernameHeader" onClick={(e) => {this.sortTable("student-list", 2)}}  value="githubUsername">Github Username</th>
+          <th id="emailHeader" onClick={(e) => {this.sortTable("student-list", 3)}}  value="email">email</th>
+        </tr></thead>
+          <StudentList students={this.state.students} />
+      </table></div>
+
+      // <tbody>
```
Comment (dearshrewdwit) [inline src/components/Students/Students.js:69]: Let's remove comments

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]
Good. Main comment is about the sort table function.

I'll have a look at the CI server tests.

**Comment by dearshrewdwit** [comment]
I ran the cypress tests @themegatree  - you're using a table to display results.

Instead, I want a list down the page of 'cards' that represent each student - Eventually I want a thumbnail photo for each student, and the pertinent info on the card, each of which is clickable that takes me to the individuals student page.

Let's keep the sort headers at the top of the page, and then sort an array of data by the header as you're doing.

File: .gitignore (hunk 1 above)
Comment (themegatree) [reply .gitignore:28]: I'll remove that, my bad

**Comment by dearshrewdwit** [comment]
@themegatree make sure you merge in the two commits I added -> this will unblock the CI tests
